ts/src/test/test.ts

```diff
@@ -231,22 +231,24 @@ export default class testMainClass extends baseMainTestClass {
         await this.importFiles (exchange);
         assert (Object.keys (this.testFiles).length > 0, 'Test files were not loaded'); // ensure test files are found & filled
         this.expandSettings (exchange);
-        const symbol = this.checkIfSpecificTestIsChosen (symbolArgv);
+        const symbol = this.checkIfSpecificTestIsChosen (exchange, symbolArgv);
         await this.startTest (exchange, symbol);
         exitScript (0); // needed to be explicitly finished for WS tests
     }
 
-    checkIfSpecificTestIsChosen (symbolArgv) {
+    checkIfSpecificTestIsChosen (exchange, symbolArgv) {
         if (symbolArgv !== undefined) {
-            const testFileNames = Object.keys (this.testFiles);
+            const testConfig = config ();
+            const tests = exchange.deepExtend (testConfig['exchange'], testConfig[exchange.id]);
+            const testNames = Object.keys (tests);
```

Comment:
review would be easier if we avoid changing non-essential things (like renaming variables) because at first, better to just change necessary parts to simplify the visual review. after merge, we can refine the  variable names etc, it would be a quick merge then.